sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/persons/PersonDirectorySteps.java

```diff
@@ -64,38 +71,136 @@ public PersonDirectorySteps(
-          webDriverHelpers.accessWebSite(LAST_CREATED_PERSON_PAGE_URL);
+          String yearOfBirth = apiState.getLastCreatedPerson().getBirthdateYYYY().toString();
           webDriverHelpers.waitForPageLoaded();
-          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(UUID_INPUT, 50);
+          webDriverHelpers.selectFromCombobox(BIRTH_YEAR_COMBOBOX, yearOfBirth);
         });
 
-    When(
-        "I search for specific person in person directory",
+    Then(
+        "I choose random value for Month of birth filter",
         () -> {
-          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(
-              SEARCH_PERSON_BY_FREE_TEXT, 30);
-          final String personUuid = EditEventSteps.person.getUuid();
-          webDriverHelpers.fillAndSubmitInWebElement(SEARCH_PERSON_BY_FREE_TEXT, personUuid);
+          String monthOfBirth = apiState.getLastCreatedPerson().getBirthdateMM().toString();
+          webDriverHelpers.selectFromCombobox(BIRTH_MONTH_COMBOBOX, monthOfBirth);
         });
 
-    When(
-        "I click on specific person in person directory",
+    Then(
+        "I choose random value for Day of birth filter",
         () -> {
-          final String personUuid = EditEventSteps.person.getUuid();
-          webDriverHelpers.clickOnWebElementBySelector(getByPersonUuid(personUuid));
+          String dayOfBirth = apiState.getLastCreatedPerson().getBirthdateDD().toString();
+          webDriverHelpers.selectFromCombobox(BIRTH_DAY_COMBOBOX, dayOfBirth);
+        });
+
+    Then(
+        "I choose present condition field from specific range",
+        () -> {
+          String presentCondition = apiState.getLastCreatedPerson().getPresentCondition();
+          webDriverHelpers.selectFromCombobox(
+              PRESENT_CONDITION, PresentCondition.getValueFor(presentCondition));
+        });
+
+    Then(
+        "I choose random value for Region",
+        () -> {
+          String regionName = apiState.getLastCreatedPerson().getAddress().getRegion();
+          webDriverHelpers.selectFromCombobox(
+              REGIONS_COMBOBOX, RegionsValues.getValueFor(regionName));
+        });
+
+    Then(
+        "I choose random value for District",
+        () -> {
+          String districtName = apiState.getLastCreatedPerson().getAddress().getDistrict();
+          webDriverHelpers.selectFromCombobox(
+              DISTRICTS_COMBOBOX, DistrictsValues.getValueFor(districtName));
+        });
+
+    Then(
+        "I choose random value for Community",
+        () -> {
+          String communityName = apiState.getLastCreatedPerson().getAddress().getCommunity();
+          webDriverHelpers.selectFromCombobox(
+              COMMUNITY_PERSON_COMBOBOX, CommunityValues.getValueFor(communityName));
         });
 
+    Then(
+        "I check the result for UID for second person in grid PERSON ID column",
+        () -> {
+          webDriverHelpers.waitUntilAListOfElementsHasText(
```

Comment:
you might need to do a pull from development because this approach with Truth is outdated